Sources/LogicKit/Term.swift

```diff
@@ -1,11 +1,16 @@
 infix operator ~=~: ComparisonPrecedence
+infix operator =>
+infix operator |-
+infix operator ⊢
+infix operator ∧
+infix operator ∨
 
 public enum Term {
 
     case `var`(String)
     case val(AnyHashable)
-    indirect case _term(name: String, arguments: [Term])
-    indirect case _rule(name: String, arguments: [Term], body: Term)
+    indirect case _term(name: Term, arguments: [Term])
```

Comment:
Why replace `String` with `Term` here?

Reply:
Switching from `String` to `Term` is used to allow users the following syntax:
```swift
let play  : Term = "play"
let mia   : Term = "mia"
play[mia]
```

If we put `String` instead, the function `fact` in `Term.swift` gives a compilation error:
```swift
    public static func fact(_ name: Term, _ arguments: Term...) -> Term {
        guard case let .val(sname) = name else { fatalError() }
        return ._term(name: sname, arguments: arguments)
    }
```
`cannot convert value of type 'AnyHashable' to expected argument type 'String'`

Maybe it can be fixed, but i do not know how.

Reply:
I don't like the of literals as term discriminators. It goes strongly against LogicKit's design. The intention of your new syntax is to be used only when the term is a literal representing a string, so as get closer to Prolog's syntax. Trying otherwise should fail, otherwise this would be technically possible:

```swift
let numbers: Term = .lit(Set([0, 1]))
let characters: Term = .lit(Set(["a", "b"]))
numbers[characters]
```

which makes little to ne sense to me.

This should solve your problem:
```swift
public subscript(terms: Term...) -> Term {
  guard case let .val(val) = self else { fatalError() }
  guard let name = val as? String else { fatalError() }
  return ._term(name: name, arguments: terms)
}
```

Reply:
Fixed